server/src/main/kotlin/com/xebia/functional/xef/server/http/routes/Routes.kt

```diff
@@ -60,36 +60,60 @@ fun Routing.routes(
             val isStream = data["stream"]?.jsonPrimitive?.boolean ?: false
 
             if (!isStream) {
-                val response = client.request("$openAiUrl/chat/completions") {
-                    headers {
-                        bearerAuth(token)
-                    }
-                    contentType(ContentType.Application.Json)
-                    method = HttpMethod.Post
-                    setBody(context)
-                }
-                call.respond(response.body<String>())
+                client.makeRequest(call, "$openAiUrl/chat/completions", context, token)
             } else {
                 runBlocking {
```

Comment:
I'm unsure if we can use something different here to `runBlocking`. /cc @nomisRev

Reply:
I'm not entirely sure of the context here but inside `post` we can call `suspend` code so we shouldn't need `runBlocking` at all 🤔

If we need to run something _async_ and return immediately (POST), then we should use `application.launch` where `application` is the property in `post` referencing to the parent server lifecycle.

Reply:
Thanks @nomisRev. I tried to replicate the behaviour of the OpenAI endpoint by using the `runBlocking`. I will try your approach 💪